go/common/sharedtypes/main.go

```diff
@@ -0,0 +1,17 @@
+package sharedtypes
```

Comment:
I'll move the `sharedtypes` package under Internal. As it is hidden from the users.  
For the `apivalidator` package, it can be imported and used externally. Go forbids this if the path contains the word `internal`.